lib/request.js

```diff
@@ -143,7 +146,9 @@ function Request (options) {
 
   if (payload && typeof payload !== 'string' && !payloadResume && !Buffer.isBuffer(payload)) {
     payload = JSON.stringify(payload)
-    this.headers['content-type'] = this.headers['content-type'] || 'application/json'
+    if ('content-type' in this.headers === false && autoFillHeaders) {
```

Comment:
This check is not the same as before, because if the user sets `headers: { 'content-type': null }`, this new check will keep the `null` value instead of filling it.

Edge case, I know, but we should deal with it

Reply:
`!this.headers['user-agent']` ?

Reply:
Yes, it supports the falsy check 👍🏽

Reply:
unfortunately it does not get rid of the error in the package if the header is explicitly set to undefined

Reply:
will this be legitimate code?
```js
  if (!this.headers['user-agent'] && autoFillHeaders) {
    this.headers['user-agent'] = 'lightMyRequest'
  } else {
    delete this.headers['user-agent']
  }
```